### ISSUE TITLE
Warnings from @iconify/tailwind4 for unrelated arbitrary value utilities

### ISSUE THREAD

**Issue by xqvvu** [schoero/eslint-plugin-readable-tailwind#377]
### Description

When `eslint-plugin-better-tailwindcss` is configured with a Tailwind CSS v4 `entryPoint` that imports `@iconify/tailwind4`, running ESLint prints `Invalid icon name` warnings for unrelated Tailwind arbitrary value utilities.

The lint result exits with code 0, but the warning output is noisy and looks like an Iconify/Tailwind dynamic plugin interaction triggered while better-tailwindcss builds or queries the Tailwind context.

I can reproduce it with Iconify's dynamic class plugin enabled, for example:

```css
@import "tailwindcss";
@plugin "@iconify/tailwind4" {
  prefix: i;
}
```

Then classes such as `border-(--site-line)`, `text-(--site-muted)`, `tracking-[0.16em]`, and `lg:grid-cols-[minmax(0,720px)_260px]` cause Iconify warnings like `Invalid icon name: "var(--site-line)"` or `Invalid icon name: "minmax(0,720px) 260px"` during `eslint .`.

Changing the Iconify plugin to clean selectors / pre-generated prefixes avoids the noise, but that requires changing the application class syntax and does not use Iconify's dynamic arbitrary class style.

### Flavor

Vue

### Code Input

```vue
<template>
  <header class="border-b border-(--site-line) bg-(--site-bg)/90">
    <span class="i-[lucide--moon] dark:i-[lucide--sun] size-4" />
  </header>

  <main class="grid gap-10 lg:grid-cols-[minmax(0,720px)_260px]">
    <p class="tracking-[0.16em] text-(--site-muted)">Blog</p>
  </main>
</template>
```

Tailwind entry point:

```css
@import "tailwindcss";
@plugin "@iconify/tailwind4" {
  prefix: i;
}

@layer components {
  .site-container {
    width: min(100% - 32px, 1040px);
    margin-inline: auto;
  }
}
```

### Expected Behavior

```text
eslint .
# no output, exit code 0
```

Arbitrary values that are not Iconify classes should not be passed to Iconify's dynamic icon resolver, or warnings from third-party Tailwind plugins should be suppressed/handled so unrelated classes do not produce noisy ESLint output.

### Actual Behavior

```text
eslint .
Invalid icon name: "var(--site-line)"
Invalid icon name: "var(--site-fg)"
Invalid icon name: "var(--site-bg)"
Invalid icon name: "var(--site-muted)"
Invalid icon name: "var(--site-accent)"
Invalid icon name: "0.16em"
Invalid icon name: "120px 1fr"
Invalid icon name: "1fr 260px"
Invalid icon name: "minmax(0,720px) 260px"
Invalid icon name: "1fr 320px"
# exit code 0
```

### Reproduction URL

_No public reproduction yet. I can create one if needed._

### ESLint Log

```shell
$ eslint .
Invalid icon name: "var(--site-line)"
Invalid icon name: "var(--site-fg)"
Invalid icon name: "var(--site-bg)"
Invalid icon name: "var(--site-muted)"
Invalid icon name: "var(--site-accent)"
Invalid icon name: "0.16em"
Invalid icon name: "120px 1fr"
Invalid icon name: "1fr 260px"
Invalid icon name: "minmax(0,720px) 260px"
Invalid icon name: "1fr 320px"
```

### ESLint Config

```ts
import betterTailwindCSS from "eslint-plugin-better-tailwindcss";

export default [
  {
    name: "plugin/tailwindcss",
    ...betterTailwindCSS.configs.recommended,
    settings: {
      "better-tailwindcss": {
        detectComponentClasses: true,
        entryPoint: "app/assets/css/main.css",
      },
    },
    rules: {
      ...betterTailwindCSS.configs.recommended.rules,
      "better-tailwindcss/enforce-consistent-class-order": "off",
      "better-tailwindcss/enforce-consistent-line-wrapping": "off",
      "better-tailwindcss/no-unknown-classes": "error",
    },
  },
];
```

### Versions

- ESLint: 9.39.4
- Parser: Vue / Nuxt ESLint config
- Plugin: eslint-plugin-better-tailwindcss 4.5.0
- Tailwind CSS: 4.3.0
- @iconify/tailwind4: 1.2.3
- Node: 24.15.0
- Package manager: pnpm 11.1.1
- OS: macOS 26 / Darwin 25.3.0 arm64

**Comment by schoero** [comment]
The warnings are logged when the `enforce-canonical-classes` rule calls [`tailwindContext.canonicalizeCandidates()`](https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/cc05c4eeee710ea20a7ac2f45af653d4635ac902/src/tailwindcss/canonical-classes.async.v4.ts#L23).

I don't really want to suppress warnings from Tailwind CSS, because those warnings could give some useful information when something else goes wrong.
There is no direct way to suppress warnings from third party plugins. We could match the warnings using regular expressions to filter "Invalid icon name" warnings, but I don't like that idea either because the warning could change with any iconify update and then it would need to be kept synchronized.

Instead, this should be solved at the root level. 

I'm not familiar with iconify. What is the root cause for those warnings? Why are values like `var(--site-line)` or `0.16em` considered icon names?

 I think it would be better to open an issue in the [iconify repository](https://github.com/iconify/iconify/issues).
